f8a_worker/workers/graph_importer_utils.py

```diff
@@ -0,0 +1,204 @@
+import os
+import requests
+import logging
+import json
+import traceback
+from f8a_worker.workers.graph_populator import GraphPopulator
+from selinon import StoragePool
+
+
+logger = logging.getLogger(__name__)
+
+# Get Variable Parameters from Environment
+GREMLIN_SERVER_URL_REST = "http://{host}:{port}".format(
+                           host=os.environ.get("BAYESIAN_GREMLIN_HTTP_SERVICE_HOST", "localhost"),
+                           port=os.environ.get("BAYESIAN_GREMLIN_HTTP_SERVICE_PORT", "8182"))
+s3 = s3_pkg = None
+
+
+def read_json_file(filename, bucket_type):
+    """Read JSON file from the data source"""
+
+    global s3, s3_pkg
+    obj = {}
+    if bucket_type == "version":
+        obj = s3.retrieve_dict(filename)
+    elif bucket_type == "package":
+        obj = s3_pkg.retrieve_dict(filename)
+
+    return obj
+
+
+def _first_key_info(first_key, bucket_type):
+    obj = {}
+    t = read_json_file(first_key, bucket_type)
+    obj["dependents_count"] = t.get("dependents_count", '-1')
+    obj["package_info"] = t.get("package_info", {})
+    obj["latest_version"] = t.get("latest_version", '-1')
+    return obj
```

Comment:
Just a question: why we do not sync data as we have them on S3 without adding additional data to them? Syncs would be more transparent and lightweight, issues will be easier to track.

Reply:
I am not following your question here. Is it, why are we creating a dict here instead of using it directly elsewhere?

Reply:
> I am not following your question here. Is it, why are we creating a dict here instead of using it directly elsewhere?

Kind of. What is the intention behind this function?

Reply:
This is more about creating the dict that graph sync understands. In this case, all the relevant meta data information about the package to be synced to graph resides in something like `<ecosystem>/<package>/<version>.json`. We just use this to modify graph sync object accordingly.